### ISSUE TITLE
Add support for certificate credentials against azuread_applications

### ISSUE THREAD

**Issue by vidarw** [terraform-providers/terraform-provider-azuread#194]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We need to be able to authenticate with private key certificates using our Azure AD applications. A popular use case for this is servers outside Azure to securely access azure resources.

### New or Affected Resource(s)

* azuread_application_certificate

### Potential Terraform Configuration

```hcl
resource "azuread_application" "example" {
  name                       = "example"
  homepage                   = "http://homepage"
  identifier_uris            = ["http://uri"]
  reply_urls                 = ["http://replyurl"]
  available_to_other_tenants = false
  oauth2_allow_implicit_flow = true
}

resource "azuread_application_certificate" "example" {
  application_id = "${azuread_application.example.id}"
  certificate = filebase64("${path.module}/public_key.pem")
}
```

### References

* https://github.com/Azure/azure-sdk-for-go/blob/master/services/graphrbac/1.6/graphrbac/applications.go#L1025
* https://github.com/terraform-providers/terraform-provider-azuread/issues/29 (related, but no the same)

**Comment by phrueegsegger** [comment]
me too..

**Comment by manicminer** [comment]
Implemented in #262
